### PR TITLE
Feat(utils): move contracts service into new folder structure wasm

### PR DESCRIPTION
**Pull request by Poafs1**

### CODE REVIEW

**Review by songwongtp — APPROVED** [review]

File: src/lib/services/types/wasm/contract.ts [hunk 1]
```diff
@@ -27,7 +21,7 @@ const zContractRest = z.object({
   contract_info: z.object({
     code_id: z.string(),
     creator: zBechAddr,
-    admin: zBechAddr,
+    admin: z.string(),
```
Comment (evilpeach) [inline src/lib/services/types/wasm/contract.ts:24]: Will it send empty string?

File: src/lib/services/types/wasm/contract.ts [hunk 2]
```diff
@@ -227,56 +156,18 @@ const zMigrationHistoriesResponseItem = z
     timestamp: val.timestamp,
     uploader: val.uploader,
   }));
-const zMigrationHistoriesResponse = z.object({
+export const zMigrationHistoriesResponse = z.object({
```
Comment (evilpeach) [inline src/lib/services/types/wasm/contract.ts:159]: apply `snakeToCamel` to line 149-158

File: src/lib/services/wasm/contract/api.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    .then(({ data }) => parseWithError(zContractQueryMsgs, data));
+
+export const getContractsByCodeId = async (
+  endpoint: string,
+  codeId: number,
+  limit: number,
+  offset: number
+): Promise<ContractsResponse> =>
+  axios
+    .get(`${endpoint}/${codeId}/contracts`, {
```
Comment (evilpeach) [inline src/lib/services/wasm/contract/api.ts:120]: apply `encode`

File: src/lib/services/types/wasm/contract.ts [hunk 4]
```diff
@@ -41,26 +35,6 @@ const zContractRest = z.object({
 });
 export type ContractRest = z.infer<typeof zContractRest>;
```
Comment (evilpeach) [inline src/lib/services/types/wasm/contract.ts:36]: ContractLcd?

File: src/lib/services/types/wasm/contract.ts (hunk 1 above)
Comment (songwongtp) [reply src/lib/services/types/wasm/contract.ts:24]: yes no admin is empty string

File: src/lib/services/types/wasm/contract.ts (hunk 4 above)
Comment (Poafs1) [reply src/lib/services/types/wasm/contract.ts:36]: @songwongtp does it already included in your revamp?

Comment (songwongtp) [reply src/lib/services/types/wasm/contract.ts:36]: duplicated, already have a type named ContractLcd

File: src/lib/services/types/wasm/contract.ts [hunk 5]
```diff
@@ -156,35 +83,50 @@ export const zContract = z
...
-    contract: zContract.nullable(),
-    contract_rest: zContractRest.nullable(),
   })
   .transform((value) => ({
-    projectInfo: value.project_info,
-    publicInfo: value.public_info,
     contract: value.contract,
     contractRest: value.contract_rest,
+    projectInfo: value.project_info,
+    publicInfo: value.public_info,
```
[the remark this answers is not in the document]
Comment (songwongtp) [reply src/lib/services/types/wasm/contract.ts:97]: done - but not leaving `contractRest` out for displaying raw snake_case key on contract details page

**Review by evilpeach — APPROVED** [review]

**Pull request merged**